spec/lib/event_publisher_spec.rb

```diff
@@ -29,78 +39,89 @@ def mock_connection(params)
+      allow(ep).to receive(:start_connection)
+      allow(ep).to receive(:create_exchanges_and_queues)
       allow(ep).to receive(:add_close_connection_handler)
       ep.create_connection
 
-      expect(ep).not_to have_received(:set_config)
+      expect(ep).not_to have_received(:start_connection)
+      expect(ep).not_to have_received(:create_exchanges_and_queues)
       expect(ep).not_to have_received(:add_close_connection_handler)
     end
   end
 
-  describe '#set_config' do
+  describe '#start_connection' do
     it 'starts a new connection' do
-
-      params = { event_conn: 'event_conn', queue_name: 'queue_name' }
-      mock_connection(params)
+      mock_connection(@params)
 
       expect(@connection).to receive(:start)
       expect(@connection).to receive(:create_channel)
       expect(@channel).to receive(:queue)
 
-      ep = EventPublisher.new(params)
+      ep = EventPublisher.new(@params)
       ep.create_connection
     end
   end
 
-  context '#publish' do
+  describe '#publishing messages' do
     setup do
-      Bunny ||= double('Bunny')
-      @params = { event_conn: 'event_conn', queue_name: 'queue_name' }
       mock_connection(@params)
-      @event_message = instance_double("EventMessage")
-      allow(@event_message).to receive(:generate_json).and_return("message")
+      @event_message = instance_double('EventMessage')
+      allow(@event_message).to receive(:generate_json).and_return('message')
 
       allow(@queue).to receive(:name).and_return(@params[:queue_name])
     end
 
-    it 'publishes a new message to the queue' do
-      unconfirmed_sets = Set.new([])
-      allow(@channel).to receive(:unconfirmed_set).and_return(unconfirmed_sets)
+    context 'unconfirmed set is empty' do
+      before(:each) do
+        @unconfirmed_sets = Set.new([])
+      end
 
-      ep = EventPublisher.new(@params)
-      expect(@exchange).to receive(:publish).with('message', routing_key: @params[:queue_name])
-      ep.publish(@event_message)
+      it 'publishes a new message to the queue' do
+        allow(@channel).to receive(:unconfirmed_set).and_return(@unconfirmed_sets)
+
+        ep = EventPublisher.new(@params)
+        expect(@exchange).to receive(:publish).with('message')
+        ep.publish(@event_message)
+      end
     end
 
-    it 'raises exception if unconfirmed set is not empty' do
-      unconfirmed_sets = Set.new([1])
-      allow(@channel).to receive(:unconfirmed_set).and_return(unconfirmed_sets)
+    context 'unconfirmed set is not empty' do
+      before(:each) do
+        @unconfirmed_sets = Set.new([1])
+      end
 
-      ep = EventPublisher.new(@params)
-      expect(@exchange).to receive(:publish).with('message', routing_key: @params[:queue_name])
-      expect{ep.publish(@event_message)}.to raise_error(/unconfirmed/)
+      it 'raises exception if unconfirmed set is not empty' do
+        allow(@channel).to receive(:unconfirmed_set).and_return(@unconfirmed_sets)
```

Comment:
Could lines 119-122 be moved into the before(:each)? (Same with 105-108)
Alternatively you could have it run before each of the contexts, then set (let) the unconfirmed_sets within each context. If I remember correctly it'd jump into the current context to get the "let" value, then use it to run the "before" code

Reply:
Next time sunshine.